core/modules/ccontrol/UserQueryAsyncResult.cc

```diff
@@ -0,0 +1,214 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "ccontrol/UserQueryAsyncResult.h"
+
+// System headers
+
+// Third-party headers
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "qmeta/Exceptions.h"
+#include "qmeta/QMeta.h"
+#include "qdisp/MessageStore.h"
+#include "sql/SqlConnection.h"
+#include "sql/SqlResults.h"
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.ccontrol.UserQueryProcessList");
+}
+
+namespace lsst {
+namespace qserv {
+namespace ccontrol {
+
+// Constructors
+UserQueryAsyncResult::UserQueryAsyncResult(QueryId queryId,
+                                           qmeta::CzarId qMetaCzarId,
+                                           std::shared_ptr<qmeta::QMeta> const& qMeta,
+                                           sql::SqlConnection* resultDbConn)
+    : UserQuery(), _qMetaCzarId(qMetaCzarId),
+      _resultDbConn(resultDbConn),
+      _messageStore(std::make_shared<qdisp::MessageStore>()) {
+
+    LOGS(_log, LOG_LVL_DEBUG, "UserQueryAsyncResult: QID=" << queryId);
+
+    // get query info from QMeta
+    try {
+        _qInfo = qMeta->getQueryInfo(queryId);
+        LOGS(_log, LOG_LVL_DEBUG, "found QMeta record: czar=" << _qInfo.czarId()
+             << " status=" << _qInfo.queryStatus() << " resultLoc=" << _qInfo.resultLocation()
+             << " msgTableName=" << _qInfo.msgTableName());
+    } catch (qmeta::QueryIdError const& exc) {
+        std::string message = "No job found for ID=" + std::to_string(queryId);
```

Comment:
This message is what users will see, I want to keep that consistent with "SHOW PROCESSLIST" column name. If you prefer I can write "QID" to the log, but that probably does not even matter — if ID is not found then it never existed and there should not be any QID in the log with the same number.